src/transformers/fiddle-embedder.js

```diff
@@ -0,0 +1,147 @@
+ */
+function transformer(tree) {
+  visitParents(tree, matchNode, visitor)
+  tree.children.unshift(importNode);
+
+  /**
+   * 
+   * @param {*} node 
+   * @param {*} ancestors 
+   * @returns { import("unist-util-visit-parents").ActionTuple }
+   */
+  function visitor(node, ancestors) {
+
+    const parent = ancestors[0];
+    const folder = node.meta;
+
+    // Find where the Fiddle code block is relative to the parent,
+    // and splice the children array to insert the embedded Fiddle
+    const index = parent.children.indexOf(node);
+    const newChildren = getFiddleAST(folder);
+    parent.children.splice(index, 1, ...newChildren)
+
+    // Return an ActionTuple [Action, Index], where
+    // Action SKIP means we want to skip visiting these new children
+    // Index is the index of the AST we want to continue parsing at.
+    return [visitParents.SKIP, index + newChildren.length]
+  }
+}
+/**
+ * From a directory in `/docs/fiddles/`, generate the AST needed
+ * for the tabbed code MDX structure.
+ * @param {string} dir 
+ */
+function getFiddleAST(dir) {
+  const files = {};
+  const children = [];
+
+  // TODO: non-alphabetic sort
+  const fileNames = fs.readdirSync(dir);
+
+  if (fileNames.length === 0) {
+    return children;
+  }
+
+  for (const file of fileNames) {
+    files[file] = fs.readFileSync(path.join(dir, file)).toString();
+  }
+
+  const tabValues = fileNames.reduce((acc, val) => {
+    return acc += `{ label: '${val}', value: '${val}', },`;
+  }, '');
+
+  let index = 0;
+
+  // Generate MDXAST structure by iterating through all files in
+  // the folder and creating <TabItem> components and code blocks
+  // for each, and bookending those with the <Tabs> component.
+
+  // The finished product should look something like:
+  // <Tabs defaultValue="id1"
+  //   values={[
+  //     {label: 'id1', value: 'id1'},
+  //     {label: 'id2', value: 'id2'}
+  //   ]}>
+  //   <TabItem value="id1">
+  //     ```js
+  //     const cow = 'say';
+  //     ```
+  //   <TabItem>
+  //   <TabItem value="id2">
+  //     ```js
+  //     const hello = 'world';
+  //     ```
+  //   <TabItem>
+  // <Tabs>
+  children.push(
+    {
+      type: 'jsx',
+      value:
+        `<Tabs defaultValue="main.js" ` +
+        `values={[${tabValues}]}>
+        <TabItem value="${fileNames[index]}">`,
+    }
+  )
+
+  while (index < fileNames.length) {
+    children.push({
+      type: 'code',
+      lang: path.extname(fileNames[index]).slice(1),
+      value: files[fileNames[index]],
+    });
+
+    index++;
+
+    if (index < fileNames.length) {
+      children.push({
+        type: 'jsx',
+        value: `</TabItem>\n<TabItem value="${fileNames[index]}">`,
+      });
+    } else {
+      children.push(
+        {
+          type: 'jsx',
+          value: `</TabItem>\n</Tabs>`,
+        },
+        {
+          type: 'jsx',
+          value: `<LaunchButton url="https://fiddle.electronjs.org/launch?target=electron/v12.0.4/${dir}"/>`
```

Comment:
Is the version part of this string configurable somehow? Seems like this would miss updates whenever new versions are released. Maybe it could reference the package json or something?

Reply:
Yep, this is just part of the prototype. I'll move the version out to a variable and add a TODO so that it's less hard-coded!